examples/nucleo_g474re/sx128x_lora/main.cpp

```diff
@@ -0,0 +1,279 @@
+// coding: utf-8
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/board.hpp>
+#include <modm/debug/logger/logger.hpp>
+#include <modm/driver/radio/sx128x.hpp>
+#include <modm/platform.hpp>
+#include <modm/processing.hpp>
+
+using Sck = GpioA5;
+using Miso = GpioA6;
+using Mosi = GpioA7;
+using SpiMaster = modm::platform::SpiMaster1;
+
+namespace
+{
+
+namespace rx
+{
+    volatile bool dio1 = false;
+    volatile bool dio2 = false;
+    volatile bool dio3 = false;
+}
+
+namespace tx
+{
+    volatile bool dio1 = false;
+    volatile bool dio2 = false;
+    volatile bool dio3 = false;
+}
+
+static constexpr modm::sx128x::LoRa::ModulationParams modulationParams = {
+    spreadingFactor : modm::sx128x::LoRa::ModulationParams::SpreadingFactor::Sf9,
+    bandwidth : modm::sx128x::LoRa::ModulationParams::Bandwidth::Bw400,
+    codingRate : modm::sx128x::LoRa::ModulationParams::CodingRate::Cr_Li_4_7
+};
+
+static constexpr modm::sx128x::LoRa::PacketParams packetParams = {
+    preambleLength : 12,
+    headerType : modm::sx128x::LoRa::PacketParams::HeaderType::Explicit,
+    payloadLength : 4,
+    crc : modm::sx128x::LoRa::PacketParams::Crc::Enable,
+    invertIq : modm::sx128x::LoRa::PacketParams::InvertIq::Standard
+};
+
+}
+
+class RxThread : public modm::sx128x, public modm::pt::Protothread
+{
+    using Reset = GpioB3;
+    using Busy = GpioB4;
+    using Dio1 = GpioB5;
+    using Dio2 = GpioB6;
+    using Dio3 = GpioB7;
+
+    using Nss = GpioD2;
+    using Hal = modm::Sx128xHalSpi<SpiMaster, Nss, Busy, Reset>;
+
+public:
+    RxThread() {}
+
+    inline bool
+    run()
+    {
+        PT_BEGIN();
+
+        Nss::setOutput(modm::Gpio::High);
+        Reset::setOutput(modm::Gpio::Low);
+        Busy::setInput(modm::platform::Gpio::InputType::PullDown);
+
+        Dio1::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio1>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { rx::dio1 = true; });
+
+        Dio2::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio2>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { rx::dio2 = true; });
+
+        Dio3::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio3>(Exti::Trigger::RisingEdge, [](uint8_t) mutable { rx::dio3 = true; });
```

Comment:
```suggestion
        Exti::connect<Dio3>(Exti::Trigger::RisingEdge, [](uint8_t) { rx::dio3 = true; });
```
I think mutable only refers to objects inside the closure, but here you're modifying s]global objects, so it's not needed.

Reply:
I have removed the mutable keyword